test-harness/terratest-extensions/modules/azure/authorizer.go

```diff
@@ -0,0 +1,29 @@
+package azure
```

Comment:
I'd like to discuss the decision to put this in the Azure package... I believe this is going to be very similar to what we need to use for integration testing and would like a single location for this. I believe you'll know best where that is.

Reply:
I put it here because the current implementation makes a lot of assumptions about the format of environment variables. Since this runs within the context of terraform, its safe to assume things like `ARM_CLIENT_ID`, `ARM_CLIENT_SECRET`, etc... exist since that is what terraform is looking for.

If we are going to make this generic, I think that it makes sense to parameterize the following:
- ARM_CLIENT_SECRET
- ARM_CLIENT_ID
- ARM_TENANT_ID
- OAuth endpoint (line 14)
- Token Endpoint (line 23)

Not sure where it would live but I agree that there is probably room for improvement here. Would like to discuss this offline w/ you.

Reply:
Refactored after discussing offline.